ksqldb-execution/src/main/java/io/confluent/ksql/execution/codegen/CodeGenRunner.java

```diff
@@ -319,10 +320,16 @@ public Void visitLambdaExpression(
 
     private void addRequiredColumn(final ColumnName columnName) {
       final Column column = schema.findValueColumn(columnName)
-          .orElseThrow(() -> new KsqlException(
-              "Cannot find the select field in the available fields."
-                  + " field: " + columnName
-                  + ", schema: " + schema.value()));
+          .orElseThrow(() -> SystemColumns.isPseudoColumn(columnName, ksqlConfig)
+              ? new KsqlException(
+              "If this is a CREATE OR REPLACE query, pseudocolumns added in newer versions of"
```

Comment:
Sorry if my previous message wasn't clear -- this additional message should be added in addition to the original message. The original message is still important for users to know, otherwise they don't know what the actual failure was.

Reply:
I think you were clear enough, I'm being a bit sloppy right now as I'm scrambling to tie up all the loose ends before end of week.